lstchain/io/dataselection.py

```diff
@@ -0,0 +1,227 @@
+        help="Range of r for event filter",
+        trait=Float(),
+        default_value=[0, 1],
+    ).tag(config=True)
+
+    wl = List(
+        help="Range of wl for event filter",
+        trait=Float(),
+        default_value=[0.01, 1],
+    ).tag(config=True)
+
+    leakage_intensity_width_2 = List(
+        help="Range for leakage_intensity_width_2 of event filter",
+        trait=Float(),
+        default_value=[0, 0.2],
+    ).tag(config=True)
+
+    fixed_gh_cut = Float(
+        help="Fixed selection cut for gh_score (gammaness)",
+        default_value=0.6,
+    ).tag(config=True)
+
+    fixed_theta_cut = Float(
+        help="Fixed selection cut for theta",
+        default_value=0.2,
+    ).tag(config=True)
+
+    fixed_source_fov_offset_cut = Float(
+        help="Fixed selection cut for source FoV offset",
+        default_value=2.83,
+    ).tag(config=True)
+
+    lst_tel_ids = List(
+        help="List of selected LST telescope ids",
+        trait=Int(),
+        default_value=[1],
+    ).tag(config=True)
+
+    def filter_cut(self, data):
+        return filter_events(data, self.event_filters())
+
+    def gh_cut(self, data):
+        return data[data["gh_score"] > self.fixed_gh_cut]
+
+    def theta_cut(self, data):
+        return data[data["theta"] < u.Quantity(self.fixed_theta_cut) * u.deg]
+
+    def true_src_fov_offset_cut(self, data):
+        return data[
+                data["true_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut) * u.deg
+            ]
+
+    def reco_src_fov_offset_cut(self, data):
+        return data[
+                data["reco_source_fov_offset"] < u.Quantity(
+                    self.fixed_source_fov_offset_cut) * u.deg
+            ]
+
+    def tel_ids_filter(self, data):
+        for i in self.lst_tel_ids:
+            data["sel_tel"] = data["tel_id"] == i
+        return data[data["sel_tel"]]
+
+    def event_filters(self):
+        """
+        Creates a dict for filter_events() function
+        """
+        self.evt_filter = {
+            "intensity": self.intensity,
+            "width": self.width,
+            "length": self.length,
+            "r": self.r,
+            "wl": self.wl,
+            "leakage_intensity_width_2": self.leakage_intensity_width_2,
+        }
+        return self.evt_filter
+
+
+class DataBinning(Component):
+    """
+    Collects information on generating energy and angular bins for
+    generating IRFs as per pyIRF requirements.
+    """
+
+    true_energy_bins = List(
+        help="Values to get true energy (TeV) binning as "
+        "[e_min, e_max, bins_per_decade]",
+        trait=Float(),
+        default_value=[0.01, 100, 5.5],
+    ).tag(config=True)
+
+    reco_energy_bins = List(
+        help="Values to get reco energy (TeV) binning as "
+        "[e_min, e_max, bins_per_decade]",
+        trait=Float(),
+        default_value=[0.01, 100, 5.5],
+    ).tag(config=True)
+
+    energy_migra_bins = List(
```

Comment:
Same as above. Even more so since the three arguments don't even have the same meaning for the different binnings!